multi-arch/samples.md

```diff
@@ -0,0 +1,157 @@
+---
+title: Samples-Non-X86
+authors:
+  - "@gabemontero"
+reviewers:
+  - "@smarterclayton"
+  - "@crawford"
+  - "@bparees"
+  - "@adambkaplan"
+approvers:
+  - "@smarterclayton"
+  - "@crawford"
+  - "@bparees"
+creation-date: 2019-09-20
+last-updated: 2019-09-20
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Sample Imagestream and Templates on non-x86 Platforms
+
+As of this writing, all samples are provided and owned by non-OpenShift 
+Development teams within Red Hat.  And none of those teams 
+
+ - provide host images on registry.redhat.io or quay.io that support PPC or Z 
+ - declare imagestreams in https://github.com/openshift/library that could reference such images
+
+OpenShift is now looking at providing support for the two non-x86 architectures from IBM in v4.x.
+
+Given that, until a resonable set of samples are provided for the IBM non-x86 architectures by either
+those other teams in Red Hat, or by IBM, the samples operator should ensure that it
+
+ - does not install any x86 imagestreams, or templates referencing those imagestreams, on non-x86 systems
+ - and by extension, always provides valid ClusterOperator statuses during install and upgrade, as this amounts to a no-op on those platforms
+
+And in case the implication is not clear, OpenShift development will not be developing or maintaining 
+any non-x86 images for these existing samples.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+As part of achieving what is described in the opening section, the samples operator needs
+to achieve the desired results within the context of how multi-arch is being achieved for 
+the entire product.
+
+Details around how mulit-arch will be achieved across the entire product are still pending,
+at least at the time of this writing for the author.
+
+In particular:
+ - will support across the entire product be staged along homogoneous / heterogeneous lines
+ - if "homogeneous only" becomes a valid option for the released product, is use of Golang's APIs for determining architecture sufficient
+ - if "heterogenous" is an option, the assumption is the list of valid architectures will be provided in a cluster-wide API object, and that samples operator installs for all architectures listed
```

Comment:
though a user could further restrict the arch's installed by updating the Architectures field in the samples operator config object to specify only a subset

Reply:
This could work.  I'd seen it as "samples operator installs for all possible architectures", which wouldn't need a config.  But maybe that's too fat?  Anyhow, we can punt this until we have homogenous working ;).